src/docfx/publish/PublishItem.cs

```diff
@@ -23,6 +23,8 @@ internal class PublishItem
 
         public string RedirectUrl { get; set; }
 
+        public bool HasError { get; set; }
```

Comment:
nit: this adds a `has_error` to every output, maybe we only include it for files that actually has errors:
```suggestion
        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
        public bool HasError { get; set; }
```

Reply:
Or ignore default value for all booleans in JsonUtiltiy.